test/arrays.js

```diff
@@ -137,6 +137,33 @@
     equal(_.sortedIndex(array, 2147483648), 2147483648, 'should work with large indexes');
   });
 
+  test('sortedIndex conforms to sortBy for undefined (#1834)', function() {
+    var sorted = _.sortBy([undefined, 1, undefined, 2]);
+    equal(_.sortedIndex(sorted, undefined, _.identity), 2);
+
+    var edgeCaseNumbers = [-Infinity, -Infinity, 0, 345345345234, Infinity, Infinity, undefined, undefined, NaN];
```

Comment:
Yeah, I know thats how I handled it in my implementation, but I don't think theres a particularly good argument for placing them one way or another, just that they should be consistently handled the same between the two methods, unless you have one.

Anyway, doing that would require changes to `_.sortBy` handling of the two values for likely minimal gain